text/0314-cdk-construct-hub.md

```diff
@@ -362,6 +367,96 @@ pipeline.
 The details of URLs to be configrued in CloudFront is available in [appendix
 "CloudFront URLs"](#cloudfront-urls).
 
+#### Backend X Frontend Compatibility
+
+As described earlier, the front-end application consumes artifacts that are stored on the backend. Since these artifacts are subject to change,
+we need to define a mechanism by which the front-end is able to detect and operate on various versions of them.
+
+To that end, we propose a generic mechanism to apply to all artifact types:
+
+Every artifact will comply to the [SemVer](https://semver.org/spec/v2.0.0.html) specification
+and include its major version in the `version` field of the schema:
+
+```json
+{
+  "version": "v1",
+  ...
+}
+```
+
+```json
+{
+  "version": "v2",
+  ...
+}
+```
+
+Every time a change in the artifact occurs, the contents of the artifact is replaced with the new version.
+This means that at any given point in time, the artifact can either be in its new version, or the old one.
+
+As a general guideline, we *prefer displaying an older version of an artifact, than to display nothing at all.*.
+To that end, the front-end application will have to support displaying all versions of the artifact.
+
+Specifically, we describe how to implement this for each type of artifact.
+
+##### Package Documentation
+
+Package documentation is created by the **Doc-Gen** function, which produces a JSON definition file:
+
+`packages/${assembly.name}/v${assembly.version}/docs-${lang}.json`
+
+Since the front-end fetches and parses this file, we'd like for it to operate on type safe interfaces that describe the schema.
+This will ensure that breaking changes will manifest as compile time errors, and force us to handle them.
+
+To that end, we create a dedicated package, called `construct-hub-docgen`, which will define this schema, and provide an API to produce it.
+The version of the schema file will be derived from the major version of the package.
```

Comment:
I'd argue the version of the schema _is_ the version of the package.

Reply:
Yeap, fair enough